test/correctness/prefetch.cpp

```diff
@@ -253,13 +260,153 @@ int test8(const Target &t) {
     return 0;
 }
 
+int test9(const Target &t) {
+    Func f("f"), g("g");
+    Var x("x"), y("y"), xo("xo"), yo("yo"), xi("xi"), yi("yi");
+
+    f(x, y) = x + y;
+    f.compute_root();
+
+    g(x, y) = f(x, y);
+    g.tile(x, y, xo, yo, xi, yi, 8, 4, TailStrategy::RoundUp)
+        .vectorize(xi)
+        .unroll(yi);
+
+    f.in()
+        .compute_at(g, xo)
+        .vectorize(x)
+        .unroll(y)
+        .prefetch(f, x, y, 123, PrefetchBoundStrategy::NonFaulting);
+
+    Module m = g.compile_to_module({}, "", t);
+    CollectPrefetches collect;
+    m.functions()[0].body.accept(&collect);
+
+    vector<vector<Expr>> expected;
+    for (int i = 0; i < 4; i++) {
+        Expr base = Variable::make(Handle(), f.name());
+        // The offset arg is a variable that is ticklish to get right, so just use a wildcard for matching
+        Expr offset = wild<int>();
+        Expr extent0 = 1;
+        Expr stride0 = get_stride(t, 4);
+        if (t.has_feature(Target::HVX)) {
+            Expr extent1 = 1;
+            Expr stride1 = wild<int>();
+            expected.push_back({base, offset, extent0, stride0, extent1, stride1});
+        } else {
+            expected.push_back({base, offset, extent0, stride0});
+        }
+    }
+    if (!check(expected, collect.prefetches)) {
+        return -1;
+    }
+    return 0;
+}
+
+int test10(const Target &t) {
+    Func f("f"), g("g");
+    Var x("x"), y("y"), xo("xo"), yo("yo"), xi("xi"), yi("yi");
+
+    f(x, y) = x + y;
+    f.compute_root();
+
+    g(x, y) = f(x, y);
+    g.tile(x, y, xo, yo, xi, yi, 8, 4, TailStrategy::RoundUp)
+        .vectorize(xi)
+        .unroll(yi);
+
+    f.in()
+        .compute_at(g, xo)
+        .split(x, xo, xi, 4)
+        .vectorize(xi)
+        .unroll(xo)
+        .reorder(xi, y, xo)
+        .unroll(y)
+        .prefetch(f, y, xo, 123 / 4, PrefetchBoundStrategy::NonFaulting);
```

Comment:
This is just a regular integer division performed at compile time, why leave it this way?

Reply:
Because it's supposed to be equivalent to the case above that prefetches 123 elements ahead in the x direction. Because this is the xo loop, the correct amount is 123/4

Maybe a comment to that effect...

Reply:
Done